lib/Language/Haskell/Stylish/Step/Imports.hs

```diff
@@ -160,6 +201,63 @@ formatImports maxCols options moduleStats rawGroup =
   forM_ group \imp -> printQualified options padNames stats imp >> newline
 
 
+--------------------------------------------------------------------------------
+groupAndFormat
+  :: Maybe Int
+  -> Options
+  -> ImportStats
+  -> [NonEmpty (GHC.LImportDecl GHC.GhcPs)]
+  -> Editor.Edits
+groupAndFormat maxCols options moduleStats groups = edits
+  where edits = Editor.changeLines block (const regroupedLines)
+        regroupedLines =
+          intercalate [""] $
+          map (formatImports maxCols options moduleStats) grouped
+
+        grouped = groupByPatterns (groupPatterns options) imports
+        imports = concatMap toList groups
+
+        block = Block
+          (GHC.srcSpanStartLine . src $ head imports)
```

Comment:
I think this is not safe in the case where `groups` is empty. Which I think might be possible if you format a file with no imports (i.e. just using the `Prelude`).